s2spy/preprocess.py

```diff
@@ -61,9 +61,36 @@ def _subtract_trend(data: Union[xr.DataArray, xr.Dataset], method: str, trend: d
     raise NotImplementedError
 
 
-def _get_climatology(data: Union[xr.Dataset, xr.DataArray]):
+def _get_climatology(data: Union[xr.Dataset, xr.DataArray], timescale: str):
     """Calculate the climatology of timeseries data."""
-    return data.groupby("time.dayofyear").mean("time")
+    # check given temporal resolution matches with data
+    if timescale == "monthly":
+        climatology = data.groupby("time.month").mean("time")
+    elif timescale == "weekly":
+        climatology = data.groupby(data["time"].dt.isocalendar().week).mean("time")
+    elif timescale == "daily" or "hourly":
```

Comment:
```suggestion
    elif timescale in ["daily", "hourly"]:
```
See https://stackoverflow.com/a/20002504

Reply:
Thanks for the suggestion. I think only for some very rare usecases, user might need `hourly` data for climatology. Also the function here is not grouped by hourly. I think it is better to just remove hourly.